src/MIDebugEngine/Engine.Impl/EngineUtils.cs

```diff
@@ -109,6 +109,33 @@ internal static bool ProcIdEquals(AD_PROCESS_ID pid1, AD_PROCESS_ID pid2)
             }
         }
 
+
+        /// <summary>
+        /// This allows console commands to be sent through the eval channel via a '-exec ' or '`' preface
+        /// </summary>
+        /// <param name="command">raw command</param>
+        /// <param name="strippedCommand">command stripped of the preface ('-exec ' or '`')</param>
+        /// <returns>true if it is a console command</returns>
+        internal static bool IsConsoleExecCmd(string command, out string prefix, out string strippedCommand)
+        {
+            prefix = string.Empty;
+            strippedCommand = string.Empty;
+            string execCommandString = "-exec ";
+            if (command.StartsWith(execCommandString, StringComparison.Ordinal))
+            {
+                prefix = execCommandString;
+                strippedCommand = command.Substring(execCommandString.Length);
+                return true;
+            }
+            else if (command[0] == '`')
+            {
+                prefix = "`";
```

Comment:
>" [](http://example.com/codeflow?start=27&length=1)

Do we need to include the whitespace here also?

Reply:
In other words, instead of using TrimStart, do we need to check for whitespace, and make the cut off the first non-whitespace character?

Reply:
There is no whitespace needed here. The prefixes are `-exec <COMMAND>` and <code>`\<COMMAND\></code>

Reply:
Are you saying in the case of `        -exec <COMMAND>`, we should still autocomplete and return with the spaces prepended? 
```
        -exec ls
        -exec list
```

Reply:
Just to make sure my question is understood. I ask because the prefix is being returned to the client as part of the match. So you are saying that VS Code is happy if the input text is `` ` pw`` (note extra spaces) but the output is <code>`pwd</code> (note the spaces went away).

Reply:
Heres the results with TrimStart() removed. 

![image](https://user-images.githubusercontent.com/3953714/135547086-a559978e-ebd2-4def-8308-6392c3870d54.png)